File: invesdwin-context-parent/invesdwin-context-scala/src/test/java/de/invesdwin/context/scala/tests/InputsAndResultsTestDecimal.scala
println("getDecimal")
if(getDecimal != null)
	throw new Exception("getDecimal already defined!")
val getDecimal = putDecimal
val getDecimalType = getDecimal.getClass()
println(getDecimalType)
println(getDecimal)
if(getDecimalType != classOf[java.lang.Double])
	throw new Exception("getDecimal not Double!")

println("getDecimalVector")
if(getDecimalVector != null)
	throw new Exception("getDecimalVector already defined!")
val getDecimalVector = putDecimalVector.asInstanceOf[Array[Double]]
val getDecimalVectorType = getDecimalVector(0).getClass()
println(getDecimalVectorType)
println(getDecimalVector)
if(getDecimalVectorType != classOf[Double])
	throw new Exception("getDecimalVector not Double!")

println("getDecimalVectorAsList")
if(getDecimalVectorAsList != null)
	throw new Exception("getDecimalVectorAsList already defined!")
val getDecimalVectorAsList = putDecimalVectorAsList.asInstanceOf[Array[Double]]
val getDecimalVectorAsListType = getDecimalVectorAsList(0).getClass()
println(getDecimalVectorAsListType)
println(getDecimalVectorAsList)
if(getDecimalVectorAsListType != classOf[Double])
	throw new Exception("getDecimalVectorAsList not Double!")

println("getDecimalMatrix")
if(getDecimalMatrix != null)
	throw new Exception("getDecimalMatrix already defined!")
val getDecimalMatrix = putDecimalMatrix.asInstanceOf[Array[Array[Double]]]
val getDecimalMatrixType = getDecimalMatrix(0)(0).getClass()
println(getDecimalMatrixType)
println(getDecimalMatrix)
if(getDecimalMatrixType != classOf[Double])
	throw new Exception("getDecimalMatrix not Double!")

println("getDecimalMatrixAsList")
if(getDecimalMatrixAsList != null)
	throw new Exception("getDecimalMatrixAsList already defined!")
val getDecimalMatrixAsList = putDecimalMatrixAsList.asInstanceOf[Array[Array[Double]]]
val getDecimalMatrixAsListType = getDecimalMatrixAsList(0)(0).getClass()
println(getDecimalMatrixAsListType)
println(getDecimalMatrixAsList)
if(getDecimalMatrixAsListType != classOf[Double])
	throw new Exception("getDecimalMatrixAsList not Double!")

File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/InputsAndResultsTestDecimal.rb
puts("getDecimal")
unless (defined?(getDecimal)).nil? || getDecimal.nil?
	raise "getDecimal already defined!"
end
getDecimal = putDecimal
getDecimalType = getDecimal.class
puts(getDecimalType)
puts(getDecimal)
if(getDecimalType != Float)
	raise "getDecimal not Float!"
end

puts("getDecimalVector")
unless (defined?(getDecimalVector)).nil? || getDecimalVector.nil?
	raise "getDecimalVector already defined!"
end
getDecimalVector = putDecimalVector
getDecimalVectorType = getDecimalVector[0].class
puts(getDecimalVectorType)
puts(getDecimalVector)
if(getDecimalVectorType != Float)
	raise "getDecimalVector not Float!"
end

puts("getDecimalVectorAsList")
unless (defined?(getDecimalVectorAsList)).nil? || getDecimalVectorAsList.nil?
	raise "getDecimalVectorAsList already defined!"
end
getDecimalVectorAsList = putDecimalVectorAsList
getDecimalVectorAsListType = getDecimalVectorAsList[0].class
puts(getDecimalVectorAsListType)
puts(getDecimalVectorAsList)
if(getDecimalVectorAsListType != Float)
	raise "getDecimalVectorAsList not Float!"
end

puts("getDecimalMatrix")
unless (defined?(getDecimalMatrix)).nil? || getDecimalMatrix.nil?
	raise "getDecimalMatrix already defined!"
end
getDecimalMatrix = putDecimalMatrix
getDecimalMatrixType = getDecimalMatrix[0][0].class
puts(getDecimalMatrixType)
puts(getDecimalMatrix)
if(getDecimalMatrixType != Float)
	raise "getDecimalMatrix not Float!"
end

puts("getDecimalMatrixAsList")
unless (defined?(getDecimalMatrixAsList)).nil? || getDecimalMatrixAsList.nil?
	raise "getDecimalMatrixAsList already defined!"
end
getDecimalMatrixAsList = putDecimalMatrixAsList
getDecimalMatrixAsListType = getDecimalMatrixAsList[0][0].class
puts(getDecimalMatrixAsListType)
puts(getDecimalMatrixAsList)
if(getDecimalMatrixAsListType != Float)
	raise "getDecimalMatrixAsList not Float!"
end
